### DISCUSSION TITLE
How to set a unique identifier for each layer

### DISCUSSION THREAD

**Discussion by boomsi** [fabricjs/fabric.js#9319] [Q&A] [answered]
I find two fieds in code. name and  data.
But i see
<img width="505" alt="image" src="https://github.com/fabricjs/fabric.js/assets/39713960/24c77306-4179-468e-bcce-73962ec4bae4">

And these two fields are not included in the exported JSON.

How to set a unique identifier that remains unchanged during export and import?

**Comment by asturur** [comment, marked as answer]
You can add to export json any properties that is in the class using the first parameter of the toObject function.

myclass.toObject(['name'])

take care that value will be exported by reference where this apply, so take care that are serializable when you go to json ot simply be careful with unwanted mutations

**Reply by boomsi** [reply]
Oh my god. Thanks very much.Very helpful.
